src/components/KTable/KTable.vue

```diff
@@ -845,11 +853,19 @@ export default defineComponent({
     }
 
     watch(() => props.searchInput, (newValue) => {
-      search(newValue)
+      if (newValue === '') {
+        search(newValue)
+      } else {
+        debouncedSearch(newValue)
+      }
     }, { immediate: true })
 
-    watch(() => [query.value, page.value, pageSize.value], () => {
-      revalidate()
+    watch(() => [query.value, page.value, pageSize.value], ([newQuery, , , oldQuery]) => {
+      if (newQuery === '' && newQuery !== oldQuery) {
+        revalidate()
+      } else {
+        debouncedRevalidate()
+      }
     }, { deep: true, immediate: true })
```

Comment:
This is a bit confusing with the empty array in the callback; could this be refactored or utilize `watchEffect` instead?

Reply:
Thanks. That's a good idea.